src/native/libs/Common/pal_utilities.h

```diff
@@ -83,7 +83,9 @@ inline static int ToFileDescriptorUnchecked(intptr_t fd)
 */
 inline static int ToFileDescriptor(intptr_t fd)
 {
+#ifndef TARGET_WASI
```

Comment:
which part of the assert is actually failing here?

Reply:
ok but this function is used in a lot more places than just those using `F_DUPFD`. I assume only a small subset run into the issue, could we make the special cases there instead of here?

Reply:
My point is that file descriptors have different valid range on wasi.
Perhaps I don't follow what do you propose.

Reply:
ok makes sense, could you please add a comment about having different valid ranges on wasi here?

I assume it makes no sense to have a wasi-specific assert since we'd essentially be checking `INT32_MIN <= fd &&  fd <= INT32_MAX` ?

Reply:
Ok, will make that note.
I have no idea what would be the upper bound on the handle value allocated by the wasi VM. Probably maximum of open files of the underlying OS ?